src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs

```diff
@@ -67,6 +73,28 @@ public void AfterCompile(AfterCompileContext context)
         {
         }
 
+
+        /// <summary>
+        /// Determines if this instance of <see cref="RazorPreCompileModule"/> should enable
+        /// compilation of views.
+        /// </summary>
+        /// <param name="context">The <see cref="BeforeCompileContext"/>.</param>
+        /// <returns><c>true</c> if views should be precompiled; otherwise <c>false</c>.</returns>
+        /// <remarks>Returns <c>true</c> if the current application is being built in <c>release</c>
+        /// configuration.</remarks>
+        protected virtual bool EnablePreCompilation(BeforeCompileContext context)
```

Comment:
We have a property on this type `GenerateSymbols` which could deserve the same treatment - i.e. change it to a method which accept the context. Although I don't see why you would need symbols for precompiled modules.